### PR TITLE
pdnssec backend-cmd

### PR DESCRIPTION
**Pull request by Habbie**
```
$ ../pdns/pdnssec --config-dir=. --config-name=gsqlite3 gsql-cmd gsqlite3 'select * from domains' 'select * from domainmetadata' | column -t -s $'\t'
== select * from domains
'1'                              'delegated.dnssec-parent.com'         ''                 ''      'NATIVE'  ''  ''
'2'                              'dnssec-parent.com'                   ''                 ''      'NATIVE'  ''  ''
'3'                              'example.com'                         ''                 ''      'NATIVE'  ''  ''
'4'                              'minimal.com'                         ''                 ''      'NATIVE'  ''  ''
'5'                              'nztest.com'                          ''                 ''      'NATIVE'  ''  ''
'6'                              'secure-delegated.dnssec-parent.com'  ''                 ''      'NATIVE'  ''  ''
'7'                              'stest.com'                           ''                 ''      'NATIVE'  ''  ''
'8'                              'test.com'                            ''                 ''      'NATIVE'  ''  ''
'9'                              'tsig.com'                            ''                 ''      'NATIVE'  ''  ''
'10'                             'wtest.com'                           ''                 ''      'NATIVE'  ''  ''
'11'                             'test.dyndns'                         ''                 ''      'NATIVE'  ''  ''
== select * from domainmetadata
'1'                              '9'                                   'TSIG-ALLOW-AXFR'  'test'
```

Comments welcome. Needs squash, perhaps docs, maybe tests (not sure this is important for what's basically a debugging tool)?

### CODE REVIEW

File: pdns/Makefile.am [hunk 1]
```diff
@@ -210,6 +210,26 @@ pdns_server_LDADD = \
...
+
+#gsql-cmd_LDFLAGS = \
+#	$(AM_LDFLAGS) \
+#	$(DYNLINKFLAGS)
+
+#gsql-cmd_LDADD = \
+#	@moduleobjects@ \
+#	@modulelibs@ \
+#	$(LIBDL)
+
```
Comment (Habbie) [inline pdns/Makefile.am]: need to remove this

File: pdns/backends/gsql/gsqlbackend.hh [hunk 2]
```diff
@@ -220,6 +220,7 @@ public:
   bool getComment(Comment& comment);
   void feedComment(const Comment& comment);
   bool replaceComments(const uint32_t domain_id, const string& qname, const QType& qt, const vector<Comment>& comments);
+  string directGSQLQuery(const string &query);
```
Comment (Habbie) [inline pdns/backends/gsql/gsqlbackend.hh]: need to decide on this name - perhaps non-gsql backends also have a use for string commands? (other SQL backends for example)

File: pdns/backends/gsql/gsqlbackend.cc [hunk 3]
```diff
@@ -1502,6 +1502,32 @@ bool GSQLBackend::replaceComments(const uint32_t domain_id, const string& qname,
   return true;
 }
 
+string GSQLBackend::directGSQLQuery(const string &query)
+{
+ try {
+   ostringstream out;
+
+   unique_ptr<SSqlStatement> stmt(d_db->prepare(query,0));
```
Comment (Habbie) [inline pdns/backends/gsql/gsqlbackend.cc]: @cmouse can you confirm that just calling the destructor on SSqlStatement when it goes out of scope is sufficient, without calling `reset` or `release`?

Comment (cmouse) [inline pdns/backends/gsql/gsqlbackend.cc]: I checked goracle, gmysql, gpgsql and gsqlite3 and they all perform cleanup in dtor.

Comment (Habbie) [inline pdns/backends/gsql/gsqlbackend.cc]: thanks!

**Comment by pieterlexis** [comment]
Looks good. Wrt docs, a few entries in the `--help` and the manpage would suffice (with the warning that the sql is passed to the db without sanitizing it)

**Comment by Habbie** [comment]
Squashed, cleaned up, documented, renamed to backend-cmd. Ready for review/merge.

**Comment by cmouse** [comment]
Markdown docs missing

**Comment by Habbie** [comment]
Markdown docs no longer missing

**Comment by cmouse** [comment]
You might want to print the command to stderr instead of stdout. This would make it easier to build automation tools around this.

**Comment by Habbie** [comment]
also done

**Comment by Habbie** [comment]
Ready for review+merge!

**Comment by cmouse** [comment]
:+1: LGTM

**Comment by pieterlexis** [comment]
lgtm to me as well

**Pull request merged**